File: database/src/main/java/com/pelisat/cesp/ceemsp/database/model/Arma.java
package com.pelisat.cesp.ceemsp.database.model;

import com.pelisat.cesp.ceemsp.database.type.ArmaStatusEnum;
import com.pelisat.cesp.ceemsp.database.type.ArmaTipoEnum;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "ARMAS")
public class Arma extends CommonModel {
    @Column(name = "EMPRESA", nullable = false)
    private int empresa;

    @Column(name = "LICENCIA_COLECTIVA", nullable = false)
    private int licenciaColectiva;

    @Column(name = "TIPO", nullable = false)
    @Enumerated(EnumType.STRING)
    private ArmaTipoEnum tipo;

    @Column(name = "CLASE", nullable = false)
    private int clase;

    @Column(name = "MARCA", nullable = false)
    private int marca;

    @Column(name = "CALIBRE", nullable = false)
    private String calibre;

    @Column(name = "BUNKER", nullable = false)
    private int bunker;

    @Column(name = "STATUS", nullable = false)
    @Enumerated(EnumType.STRING)
    private ArmaStatusEnum status;

    @Column(name = "SERIE", nullable = false)
    private String serie;

    @Column(name = "MATRICULA", nullable = false)
    private String matricula;

    @Column(name = "PERSONAL")
    private Integer personal;

    @Column(name = "INCIDENCIA")
    private Integer incidencia;

    @Column(name = "MOTIVO_BAJA")
    private String motivoBaja;

    @Column(name = "OBSERVACIONES_BAJA")
    private String observacionesBaja;

    @Column(name = "DOCUMENTO_FUNDATORIO_BAJA")
    private String documentoFundatorioBaja;

    @Column(name = "FECHA_BAJA")
    private LocalDate fechaBaja;
}

File: ceemsp-frontend/src/app/_models/Arma.ts
import {Deserializable} from "./Deserializable";
import EmpresaLicenciaColectiva from "./EmpresaLicenciaColectiva";
import ArmaClase from "./ArmaClase";
import ArmaMarca from "./ArmaMarca";
import EmpresaDomicilio from "./EmpresaDomicilio";
import Persona from "./Persona";
import Incidencia from "./Incidencia";
import Empresa from "./Empresa";

export default class Arma implements Deserializable {
  id: number;
  uuid: string;
  licenciaColectiva: EmpresaLicenciaColectiva;
  tipo: string;
  clase: ArmaClase;
  marca: ArmaMarca;
  calibre: string;
  bunker: EmpresaDomicilio;
  status: string;
  serie: string;
  matricula: string;
  personal: Persona;
  eliminado: boolean;
  eliminadoIncidencia: boolean;
  fechaCreacionIncidencia: string;
  fechaEliminacionIncidencia: string;
  incidencia: Incidencia;
  personalAsignado: Persona;

  motivoBaja: string;
  observacionesBaja: string;
  documentoFundatorioBaja: string;
  fechaBaja: string;
  fechaCreacion: string;
  razonBajaIncidencia: string;
  empresa: Empresa;

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }
}
